dataclients/kubernetes/kube.go

```diff
@@ -733,31 +733,41 @@ func catchAllRoutes(routes []*eskip.Route) bool {
 //
 // where for a weight of 1.0 no Traffic predicate will be generated.
 func computeBackendWeights(backendWeights map[string]float64, rule *rule) {
-	type sumCount struct {
-		sum   float64
-		count int
+	type pathInfo struct {
+		sum        float64
+		lastActive *backend
+		count      int
 	}
 
 	// get backend weight sum and count of backends for all paths
-	pathSumCount := make(map[string]*sumCount)
+	pathInfos := make(map[string]*pathInfo)
 	for _, path := range rule.Http.Paths {
-		sc, ok := pathSumCount[path.Path]
+		sc, ok := pathInfos[path.Path]
 		if !ok {
-			sc = &sumCount{}
-			pathSumCount[path.Path] = sc
+			sc = &pathInfo{}
+			pathInfos[path.Path] = sc
 		}
 
 		if weight, ok := backendWeights[path.Backend.ServiceName]; ok {
 			sc.sum += weight
+			if weight > 0 {
```

Comment:
why do you need this check?

Reply:
To skip backends with weights explicitly set to 0.0.